src/main/java/com/blamejared/contenttweaker/fluids/FluidBuilder.java

```diff
@@ -0,0 +1,140 @@
+        this.isMolten = isMolten;
+        this.color = color;
+        builder.sound(isMolten ? SoundEvents.ITEM_BUCKET_FILL_LAVA : SoundEvents.ITEM_BUCKET_FILL, isMolten ? SoundEvents.ITEM_BUCKET_EMPTY_LAVA : SoundEvents.ITEM_BUCKET_EMPTY);
+    }
+
+    /**
+     * The light-level emitted by the fluid
+     *
+     * default value is 0
+     */
+    @ZenCodeType.Method
+    public FluidBuilder luminosity(int luminosity) {
+        builder.luminosity(luminosity);
+        return this;
+    }
+
+    /**
+     * How fast you can walk in the fluid?
+     *
+     * default value is 1000
+     */
+    @ZenCodeType.Method
+    public FluidBuilder density(int density) {
+        builder.density(density);
+        return this;
+    }
+
+    /**
+     * The Fluid's temperature
+     *
+     * default value is 300
+     */
+    @ZenCodeType.Method
+    public FluidBuilder temperature(int temperature) {
+        builder.temperature(temperature);
+        return this;
+    }
+
+    /**
+     * How quickly the fluid spreads
+     *
+     * default value is 1000
+     */
+    @ZenCodeType.Method
+    public FluidBuilder viscosity(int viscosity) {
+        builder.viscosity(viscosity);
+        return this;
+    }
+
+    /**
+     * Is the fluid gaseous (flows upwards instead of downwards)?
+     */
+    @ZenCodeType.Method
+    public FluidBuilder gaseous() {
+        builder.gaseous();
+        return this;
+    }
+
+    @Override
+    public void build(ResourceLocation location) {
+        String path = location.getPath();
+        IIsCotFluid stillFluid = new CoTStillFluid(location);
+        IIsCotFluid flowingFluid = new CoTFlowingFluid(location);
+
+        ForgeFlowingFluid.Properties properties = new ForgeFlowingFluid.Properties(stillFluid, flowingFluid, builder);
+        AbstractBlock.Properties fluidBlockProperties = AbstractBlock.Properties.create(isMolten ? Material.WATER : Material.LAVA)
+                .doesNotBlockMovement()
+                .hardnessAndResistance(100.0f)
+                .noDrops();
+        FlowingFluidBlock fluidBlock = new FlowingFluidBlock(stillFluid, fluidBlockProperties);
+        CoTFluidBucketItem bucketItem = new CoTFluidBucketItem(stillFluid, new Item.Properties().maxStackSize(1).containerItem(Items.BUCKET).group(ItemGroup.MISC));
+
+        bucketItem.setRegistryName(path + "_bucket");
+        fluidBlock.setRegistryName(stillFluid.getRegistryNameNonNull());
+        properties.block(() -> fluidBlock);
+        properties.bucket(() -> bucketItem);
+        stillFluid.setFluidBlock(fluidBlock);
+        flowingFluid.updateFluid((FlowingFluid) new ForgeFlowingFluid.Flowing(properties).setRegistryName(flowingFluid.getRegistryNameNonNull()));
+        stillFluid.updateFluid((FlowingFluid) new ForgeFlowingFluid.Source(properties).setRegistryName(stillFluid.getRegistryNameNonNull()));
+
+        VanillaFactory.queueFluidForRegistration(stillFluid);
+        VanillaFactory.queueFluidForRegistration(flowingFluid);
```

Comment:
It may confuse the user when there are two registry messages (one for flowing, one for still), but that should be fine.